### DISCUSSION TITLE
Setting up aws rekognition with AI tool

### DISCUSSION THREAD

**Discussion by garhun** [VorlonCD/bi-aidetection#224] [Q&A]
I am setting up AI on my Blue Iris system, using the AI tool ver 2.0.942.7743 with Deepstack for Windows CPU.2020.12Beta. AI tool seems to be working well,, getting my trigger images from BI as it should. However, I cannot get Deepstack to start up completely. Log indicates that 2 python files are not running. Since I have not been able to fix that, I noticed the AI Tool seems to offer a connection to aws reckognition in the Server URL setup, so I am investigating how to accomplish this connection in place of Deepstack if that is possible. I have set up my user account in AWS, and can generate user access keys, but don't know how to implement that into the AI Tool..There is a pop up in AI Tool, AI Servers,that says to download root key and place it in AITOOL_settings folder, but I don't find a root key anywhere in my AWS access management page. Just to note, I am not a software developer, but a home automation enthusiast, and would appreciate any direction you can give me toward solving either or both of my hangups. I really want to add this feature to my system.
I'll try attaching a few screenshots that may be helpful.

![image](https://user-images.githubusercontent.com/16842532/113339216-2b3e9b00-92f8-11eb-989d-5b5070afeee4.png)
![image](https://user-images.githubusercontent.com/16842532/113339010-e1ee4b80-92f7-11eb-9381-a8207e83609b.png)
![image](https://user-images.githubusercontent.com/16842532/113339104-ffbbb080-92f7-11eb-8acd-e6fbf6f04794.png)

**Comment by VorlonCD** [comment]
Try a later version of deepstack:
https://github.com/johnolafenwa/DeepStack/releases/tag/0.3-beta

For AWS go to the last url in the message (also in the log), pick Access Keys, **delete** the existing access key and create a new one.  During that process you will be prompted to save rootkey.csv.  Put it in the Settings folder.

Keep in mind AWS Recognition is limited to 5000 images a month before they start charging.

**Comment by garhun** [comment]
Thanks for your response.  I appreciate your work. And will try those suggestions tonight. Gary Sent from my T-Mobile 5G Device
-------- Original message --------From: VorlonCD ***@***.***> Date: 4/2/21  9:46 AM  (GMT-05:00) To: VorlonCD/bi-aidetection ***@***.***> Cc: garhun ***@***.***>, Author ***@***.***> Subject: Re: [VorlonCD/bi-aidetection] Setting up aws rekognition with AI tool
  (#224) 
Try a later version of deepstack:
https://github.com/johnolafenwa/DeepStack/releases/tag/0.3-beta
For AWS go to the last url in the message (also in the log), pick Access Keys, delete the existing access key and create a new one.  During that process you will be prompted to save rootkey.csv.  Put it in the Settings folder.
Keep in mind AWS Recognition is limited to 5000 images a month before they start charging.

**Comment by garhun** [comment]
Thanks for your help. With it, I was able to get the aws service working. When I tried the suggested version of Deepstack, however, I still had the same result. Deepstack failed to run, giving the log message about 0 of 2 python files not starting. It would be nice to get that working, so if you have any other suggestions, please advise. 

Thanks again.

Gary

**Comment by kozoke** [comment]
I've had good luck running the Deepstack container both on the windows
machine as well as my unraid server.
Have you explored running docker on Windows and launching the Deepstack
container?
https://www.deepstack.cc/install/docker
(Activation no longer required)

**Comment by garhun** [comment]
Thanks for your suggestion.  I have thought about trying the docker version,  but was hesitant since the developer said no docker needed for Windows.  I have no experience with docker installations, but suppose I can try it if the instructions are clear. I will give it a try. Thanks again. GarySent from my T-Mobile 5G Device
-------- Original message --------From: kozoke ***@***.***> Date: 4/2/21  8:07 PM  (GMT-05:00) To: VorlonCD/bi-aidetection ***@***.***> Cc: garhun ***@***.***>, Author ***@***.***> Subject: Re: [VorlonCD/bi-aidetection] Setting up aws rekognition with AI tool
  (#224) 

I've had good luck running the Deepstack container both on the windows

machine as well as my unraid server.

Have you explored running docker on Windows and launching the Deepstack

container?

https://www.deepstack.cc/install/docker

(Activation no longer required)

**Comment by VorlonCD** [comment]
Try running only one copy/port of deepstack on the aitool deepstack tab.   The developer says more than one instance can run but I always seem to have trouble.   GPU version is worse than CPU version.  But some still run multiple copies successfully.
Are you running AITOOL "as administrator"?   If so, try disabling.
Are you at a fairly recent version of Windows with all windows updates?
Tried disabling antivirus?
Hit RESET button on deepstack tab to reset all of its settings, then start it again.
If still failing hit the STDERR button and send any text that is in the file.
Also send actual log entries of deepstack starting.  (you can ctrl-c selected entries in the log tab)

**Reply by garhun** [reply]
Thanks for responding again. I have tried each of your recommendations above, and still no success. I am attaching relevant screenshots that you requested. I am not running AITool as admin, and only one instance of Deepstack. I did not find any relevant data in the Win event logs. I have definitely reached my depth here, and appreciate your work.

![Screenshot_Log_first 2021-04-03 195131](https://user-images.githubusercontent.com/16842532/113494462-65559b80-94b6-11eb-8b1e-77950d7f58bb.png)
![Screenshot_Log_2 2021-04-03 195254](https://user-images.githubusercontent.com/16842532/113494471-6dadd680-94b6-11eb-97e7-0d6e4b006af9.png)



![Screenshot_Stdrr 2021-04-03 173830](https://user-images.githubusercontent.com/16842532/113494243-9cc34880-94b4-11eb-869e-c5d1ce62186d.png)

![WinVerScreenshot 2021-04-03 193523](https://user-images.githubusercontent.com/16842532/113494250-a77ddd80-94b4-11eb-81f7-d24126a4842f.png)

**Reply by VorlonCD** [reply]
Dig around up on the Deepstack forums.  I see someone else had the same issue:
https://forum.deepstack.cc/t/deepstack-windows-cpu/776

A quick googling says it may be a visual c++ runtime issue.    Try using [PATCHMYPC](https://patchmypc.com/freeupdater/PatchMyPC.exe) to install all the latest runtimes in this section:
![image](https://user-images.githubusercontent.com/7512113/113514957-8a422100-953f-11eb-9cfa-847c5e39f725.png)

Also maybe there is some kind of version conflict with another version of PYTORCH installed on your machine?    I would typically assume that pytorch installed automatically in the deepstack folder is isolated, but maybe not.

Run [Dependencies](https://github.com/lucasg/Dependencies/releases/download/v1.10/Dependencies_x64_Release.zip) > DependenciesGUI.exe, and  then drag and drop "C:\DeepStack\windows_packages\torch\lib\asmjit.dll" into the window.   It should give you an error of some kind.

**Reply by VorlonCD** [reply]
@garhun - any luck with this?

**Reply by garhun** [reply]
I had no success doing the above. I also removed and reinstalled Deepstack. Searched for any other Python files on computer. Still getting same errors at start up. Could not find any solutions on Deepstack forum,  though did see other posts with same issue.

**Reply by VorlonCD** [reply]
can send screenshot of how the "Dependencies" program looks when you load ASMJIT.DLL into it?

**Reply by garhun** [reply]
I know how to screen shot, but I don't know how to load that DLL, or what Dependencies program is. I did look at the file structure of the Deepstack installation, and in the Interpreter file, I found some Python stuff, including a zipped folder that I though was odd. But maybe not. Ill attach a screenshot. 
![Screenshot_Python 2021-04-06 145832](https://user-images.githubusercontent.com/16842532/113764968-ad103900-96e9-11eb-917d-e9fded2b4be2.png)

**Reply by garhun** [reply]
My apologies. I just noticed that your first response had a link to Dependencies. I am trying that now.

**Reply by garhun** [reply]
screenshot of how the "Dependencies" program looks when you load ASMJIT.DLL into it?
![Screenshot_Dependencies 2021-04-06 165439](https://user-images.githubusercontent.com/16842532/113777489-25cac180-96f9-11eb-9e4f-48f0890782ec.png)

**Reply by VorlonCD** [reply]
Looks same as mine.  If I start expanding each DLL I see a few errors, but mine is working.

Try uninstalling the 2015 c++ runtimes from control panel then re-installing via this universal installer  Specifically I believe VCRUNTIME140.DLL belongs to **Visual C++ 2015**.   And I just noticed that PATCHMYPC doesnt HAVE 2015 in its list!
https://www.techpowerup.com/download/visual-c-redistributable-runtime-package-all-in-one/

If you have python installed (in control panel), uninstall it and test.

Does asmjit.dll exist anywhere else on your drive outside of the deepstack folder?    Does PYTHON.EXE exist outside the deepstack folder?  (dont just delete the other copies unless you are sure you know what you are doing and make backups)
I use VoidTools Everything search for a quick drive scan of any files. https://www.voidtools.com/

**Reply by garhun** [reply]
Good news! I could not find the C++ 2015 on my computer (there were several other years, but no 2015), so I just downloaded the redistributed packages that you said. That was after I had deleted some old python files that were on my system from a Home assistant installation that I had forgotten about. Now Deepstack starts and runs as it should. Thanks again for your invaluable work and assistance solving my problem.

**Reply by VorlonCD** [reply]
Ok, great!
